src/main/kotlin/calculator/Calculator.kt

```diff
@@ -0,0 +1,46 @@
+package calculator
+
+class Calculator {
+    fun plus(a: Long, b: Long) = a + b
+    fun minus(a: Long, b: Long) = a - b
+    fun multiply(a: Long, b: Long) = a * b
+    fun divide(a: Long, b: Long) = a / b
+
+    fun calculate(formula: String?): Long {
+        if (formula.isNullOrBlank())
+            throw IllegalArgumentException()
+
+        val iter = formula.split(" ").iterator()
+            .takeIf { it.hasNext() }
+            ?: throw IllegalArgumentException()
+
+        var left = iter.nextLong()
+
+        while (iter.hasNext()) {
+            val operator = iter.nextOperator()
+            val right = iter.nextLong()
+
+            left = when (operator) {
+                '+' -> plus(left, right)
+                '-' -> minus(left, right)
+                '*' -> multiply(left, right)
+                '/' -> divide(left, right)
+                else -> throw IllegalArgumentException()
+            }
+        }
+
+        return left
+    }
+
+    private fun Iterator<String>.nextLong() = runCatching {
+        next().toLong()
+    }.getOrElse { throw IllegalArgumentException() }
+
+    private fun Iterator<String>.nextOperator() = runCatching {
+        next()[0]
+    }.getOrElse { throw IllegalArgumentException() }
```

Comment:
확장함수와 runCatching 활용 좋습니다. 👍 
개인적으로는 예외를 감싸서 다른 예외로 전환할때는 원래 예외를 함께 전달하는것이 좋다고 생각합니다. 이유는 종종 디버깅할 때 유용하게 때문이에요.
```suggestion
    }.getOrElse { throw IllegalArgumentException(it) }
```